From 10830b1c3121f74085081bc9a92ff4e24527dbc9 Mon Sep 17 00:00:00 2001
From: Adrian Fiergolski <adrian.fiergolski@cern.ch>
Date: Tue, 31 Jan 2017 16:59:17 +0100
Subject: [PATCH] Add SPI interface of the CLICpix2 chip to the device tree.

---
 outputs/dts/pl.dtsi | 9 +++++++++
 1 file changed, 9 insertions(+)

diff --git a/outputs/dts/pl.dtsi b/outputs/dts/pl.dtsi
index 0a06fcb..23bcef0 100644
--- a/outputs/dts/pl.dtsi
+++ b/outputs/dts/pl.dtsi
@@ -21,6 +21,15 @@
 			reg = <0x41e00000 0x10000>;
 			xlnx,num-ss-bits = <0x1>;
 			xlnx,spi-mode = <0>;
+			spidev@0 {
+				 #address-cells = <1>;
+				 #size-cells = <0>;
+
+				 compatible = "caribou,clicpix2";
+				 reg = <0>;
+				 spi-max-frequency = <100000000>;
+			};
+			
 		};
 	};
 };
-- 
2.9.3

