From 74510be1751406a13e4092f2020beb370915e532 Mon Sep 17 00:00:00 2001
From: Adrian Fiergolski <adrian.fiergolski@cern.ch>
Date: Thu, 16 Mar 2017 18:08:06 +0100
Subject: [PATCH 2/2] Add CLICpix2 SPI interface to the device tree.

Signed-off-by: Adrian Fiergolski <adrian.fiergolski@cern.ch>
---
 outputs/dts/pl.dtsi | 7 ++++++-
 1 file changed, 6 insertions(+), 1 deletion(-)

diff --git a/outputs/dts/pl.dtsi b/outputs/dts/pl.dtsi
index 7c7e56f..d5ad863 100644
--- a/outputs/dts/pl.dtsi
+++ b/outputs/dts/pl.dtsi
@@ -12,12 +12,26 @@
 		compatible = "simple-bus";
 		ranges ;
 		AXI_SPI_QUAD_staic_caribou_top_axi_quad_spi_0_0_0: staic_caribou_top_axi_quad_spi_0_0@43c00000 {
-			compatible = "xlnx,staic-caribou-top-axi-quad-spi-0-0-1.0";
+			compatible = "xlnx,xps-spi-2.00.a";
 			interrupt-parent = <&intc>;
 			interrupts = <0 29 1>;
 			reg = <0x43c00000 0x10000>;
 			xlnx,sck-ratio1 = <0x1>;
 			xlnx,use-startup-int = <0x0>;
+			bits-per-word = <16>;
+			num-cs = <0x1>;
+			xlnx,num-ss-bits = <0x1>;
+			xlnx,spi-mode = <0>;
+			fifo-size = <256>;
+			spidev@0 {
+				 #address-cells = <1>;
+				 #size-cells = <0>;
+
+				 compatible = "caribou,clicpix2";
+				 reg = <0>;
+				 spi-max-frequency = <100000000>;
+				 spi-bits-per-word = <16>;
+			};
 		};
 	};
 };
-- 
2.9.3

