From 53344a2bec1aca4d3b180d5eb3281cf365cbc264 Mon Sep 17 00:00:00 2001
From: Adrian Fiergolski <adrian.fiergolski@cern.ch>
Date: Tue, 29 Aug 2017 16:15:50 +0200
Subject: [PATCH 2/2] Add CLICpix2 SPI interface to the device tree.

Signed-off-by: Adrian Fiergolski <adrian.fiergolski@cern.ch>
---
 outputs/dts/pl.dtsi | 9 +++++++++
 1 file changed, 9 insertions(+)

diff --git a/outputs/dts/pl.dtsi b/outputs/dts/pl.dtsi
index 143d6f6..6fd52db 100644
--- a/outputs/dts/pl.dtsi
+++ b/outputs/dts/pl.dtsi
@@ -34,6 +34,15 @@
 				reg = <0x43c30000 0x10000>;
 				xlnx,num-ss-bits = <0x1>;
 				xlnx,spi-mode = <0>;
+				spidev@0 {
+					  #address-cells = <1>;
+					  #size-cells = <0>;
+
+					  compatible = "caribou,clicpix2";
+				 	  reg = <0>;
+				 	  spi-max-frequency = <100000000>;
+				 	  spi-bits-per-word = <16>;
+				};
 			};
 		};
 		Caribou_control_0: Caribou_control@43c00000 {
-- 
2.14.1

