From 141309853152998289bce053767e6ece9c677488 Mon Sep 17 00:00:00 2001
From: Adrian Fiergolski <adrian.fiergolski@cern.ch>
Date: Tue, 29 Aug 2017 16:15:50 +0200
Subject: [PATCH 2/2] Add CLICpix2 SPI interface to the device tree.

Signed-off-by: Adrian Fiergolski <adrian.fiergolski@cern.ch>
---
 outputs/dts/pl.dtsi | 10 ++++++++++
 1 file changed, 10 insertions(+)

diff --git a/outputs/dts/pl.dtsi b/outputs/dts/pl.dtsi
index 791cdb6..bcec91c 100644
--- a/outputs/dts/pl.dtsi
+++ b/outputs/dts/pl.dtsi
@@ -33,6 +33,16 @@
 			reg = <0x43c00000 0x10000>;
 			xlnx,num-ss-bits = <0x1>;
 			xlnx,spi-mode = <0>;
+			spidev@0 {
+				 #address-cells = <1>;
+				 #size-cells = <0>;
+
+				 compatible = "caribou,clicpix2";
+				 reg = <0>;
+				 spi-max-frequency = <100000000>;
+				 spi-bits-per-word = <16>;
+			};
+
 		};
 		Caribou_control_0: Caribou_control@43c30000 {
 			compatible = "xlnx,Caribou-control-1.0";
-- 
2.11.0

