From 8460386bc42e902bd2cb424b27cc7ec307edd6d5 Mon Sep 17 00:00:00 2001
From: Adrian Fiergolski <adrian.fiergolski@cern.ch>
Date: Mon, 28 Aug 2017 17:29:10 +0200
Subject: [PATCH 4/5] Fix SPI xilinx driver to respect bits_per_word.

Signed-off-by: Adrian Fiergolski <adrian.fiergolski@cern.ch>
---
 drivers/spi/spi-xilinx.c | 6 +++---
 1 file changed, 3 insertions(+), 3 deletions(-)

diff --git a/drivers/spi/spi-xilinx.c b/drivers/spi/spi-xilinx.c
index 973e26f096a9..a5431b8a3bc2 100644
--- a/drivers/spi/spi-xilinx.c
+++ b/drivers/spi/spi-xilinx.c
@@ -136,7 +136,7 @@ static void xspi_read_rx_fifo_##size(struct xilinx_spi *xqspi)		\
 	int count = (xqspi->bytes_to_receive > xqspi->buffer_size) ?	\
 			xqspi->buffer_size : xqspi->bytes_to_receive;	\
 	u32 data;							\
-	for (i = 0; i < count; i += (size/8)) {				\
+	for (i = 0; i < count/sizeof( type ); i++) {		        \
 		data = readl_relaxed(xqspi->regs + XSPI_RXD_OFFSET);	\
 		if (xqspi->rx_ptr)					\
 			((type *)xqspi->rx_ptr)[i] = (type)data;	\
@@ -163,7 +163,7 @@ static void xspi_fill_tx_fifo_##size(struct xilinx_spi *xqspi)		\
 	u32 data = 0;							\
 	for (i = 0; i < count; i += (size/8)) {				\
 		if (xqspi->tx_ptr)					\
-			data = (type)((u8 *)xqspi->tx_ptr)[i];		\
+		  data = * ((type *) ( (u8 *) xqspi->tx_ptr +i));	\
 		writel_relaxed(data, (xqspi->regs + XSPI_TXD_OFFSET));	\
 	}								\
 	xqspi->bytes_to_transfer -= count;				\
@@ -780,7 +780,7 @@ static int xilinx_spi_probe(struct platform_device *pdev)
 	master->transfer_one = xspi_start_transfer;
 	master->prepare_transfer_hardware = xspi_prepare_transfer_hardware;
 	master->unprepare_transfer_hardware = xspi_unprepare_transfer_hardware;
-	master->bits_per_word_mask = SPI_BPW_MASK(8);
+	master->bits_per_word_mask = SPI_BPW_MASK( bits_per_word );
 	master->mode_bits = SPI_CPOL | SPI_CPHA | SPI_CS_HIGH;
 
 	xspi->bytes_per_word = bits_per_word / 8;
-- 
2.11.0

