From ae983bedcab2461259defd3e97f9a513597a5014 Mon Sep 17 00:00:00 2001
From: Adrian Fiergolski <adrian.fiergolski@cern.ch>
Date: Fri, 24 Nov 2017 17:23:24 +0100
Subject: [PATCH] SPI device driver lets the SPI core to drive the SS.

Signed-off-by: Adrian Fiergolski <adrian.fiergolski@cern.ch>
---
 drivers/spi/spi-xilinx.c | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/drivers/spi/spi-xilinx.c b/drivers/spi/spi-xilinx.c
index 8f864e6fd995..0e646cb4dfaa 100644
--- a/drivers/spi/spi-xilinx.c
+++ b/drivers/spi/spi-xilinx.c
@@ -255,10 +255,10 @@ static void xspi_init_hw(struct xilinx_spi *xspi)
 	/* Deselect the slave on the SPI bus */
 	xspi->write_fn(0xffff, regs_base + XSPI_SSR_OFFSET);
 	/*
-	 * Disable the transmitter, enable Manual Slave Select Assertion,
+	 * Disable the transmitter,
 	 * put SPI controller into master mode, and enable it
 	 */
-	xspi->write_fn(XSPI_CR_MANUAL_SSELECT |	XSPI_CR_MASTER_MODE |
+	xspi->write_fn(XSPI_CR_MASTER_MODE |
 		XSPI_CR_ENABLE | XSPI_CR_TXFIFO_RESET |	XSPI_CR_RXFIFO_RESET,
 		regs_base + XSPI_CR_OFFSET);
 }
-- 
2.14.1

